### ISSUE TITLE
.ps1 shims don't work as expected with PowerShell 7 chain operators

### ISSUE THREAD

**Issue by ocitrev** [ScoopInstaller/Scoop#3850]
With PowerShell 7 you can use chain operators. Without a proper value in `$?` it does not work as expected.

When I run this command line:
```
PS> should-fail || echo "Failed"
```
I get no output.
With current shims, PowerShell always see success (with automatic variable `$?`) when running .ps1 shims.

As a workaround, I can add .exe to the name of the command.  This will bypass the .ps1 shim and work as expected.
```
PS> should-fail.exe || echo "Failed"
Failed
```

**Comment by ocitrev** [comment]
In case any one wants to try should-fail.exe, here is the source code.

```C++
int main() { return 1; }
```

**Comment by rashil2000** [comment]
Merged in develop branch - #4543
